### DISCUSSION TITLE
Roborock S7 Pro Ultra

### DISCUSSION THREAD

**Discussion by petrusb83** [rytilahti/python-miio#1667] [General]
I have a S7 pro ultra. It worked fine with this integration until 6 January, then I got the issue below and now it is unavailable. My S5 still works without any problems. I also checked and the S7 still works fine in the Mi Home app.

Logger: miio.device
Source: components/xiaomi_miio/__init__.py:233
First occurred: January 6, 2023 at 10:39:57 PM (1 occurrences)
Last logged: January 6, 2023 at 10:39:57 PM

Found an unsupported model 'roborock.vacuum.a62' for class 'RoborockVacuum'. If this is working for you, please open an issue at https://github.com/rytilahti/python-miio/

**Comment by rytilahti** [comment]
Afaik, there have been no major changes that should cause it not to work. The warning you posted just informs that this model is not defined for roborockvacuum class, but does not affect the functionality whatsoever.

Do you see any other errors related to that vacuum? Has something changed in your network setup?

Could you also try the miiocli tool as described in the readme of this project to verify if it's working?

**Reply by petrusb83** [reply]
I have the following errors as well:

_Logger: miio.miioprotocol
Source: components/xiaomi_miio/__init__.py:242
First occurred: 4:06:42 AM (1 occurrences)
Last logged: 4:06:42 AM

error while reading discover results: Error in path (parsing) -> header parsing expected 8497 but parsed 9216

Logger: homeassistant.components.xiaomi_miio
Source: helpers/update_coordinator.py:168
Integration: Xiaomi Miio (documentation, issues)
First occurred: January 7, 2023 at 3:08:48 AM (12 occurrences)
Last logged: 12:46:32 PM

Timeout fetching Roborock Vacuum data
Timeout fetching Roborock S7 Pro Ultra data_

I did not change anything on my network settings. Will update HA and restart. Maybe that will fix it.

**Reply by rytilahti** [reply]
That error message is coming from https://github.com/rytilahti/python-miio/blob/master/miio/miioprotocol.py#L138 (caused by invalid data being fed to https://github.com/rytilahti/python-miio/blob/master/miio/miioprotocol.py#L121). So, for some reason, the device is responding with something unexpected.

Have you upgraded its firmware recently? I cannot think of anything besides that or a network issue that could be causing this error I'm seeing for the first time ever.

**Reply by petrusb83** [reply]
I checked, no recent firmware updates. Strange. Maybe something on their side with this specific model. My S5 is still fine.
